### DISCUSSION TITLE
branch convention에 대한 의견

### DISCUSSION THREAD

**Discussion by DonghyunNa** [QuesDevTeam/Git-Convention#26] [Ideas]
1. 기본 브랜치 변경에 대한 논의

- [github notice](https://github.blog/changelog/2020-10-01-the-default-branch-for-newly-created-repositories-is-now-main/)
- [기사](https://zdnet.co.kr/view/?no=20200921101131)

기본 브랜치가 `master` 에서 `main`으로 변경됬습니다.
main, master 모두 naming에 큰 의미가 없다고 생각이 들어, 저희도 앞으로 생성할 repository에 대해서는 `main` 을 사용하는게 어떨까 싶습니다.
(github에서 제공하는 기본 값이므로)

2. 컨벤션 상 사용하는 branch의 변경

- [참고](https://github.com/QuesDevTeam/Git-Convention/tree/master/branch#branches)
```
우리가 사용하는 브랜치들은 다음과 같습니다.
master, develop, feature, enhancement, fix, hotfix
```

우선 `enhancement`에 할당하는 작업의 성격을 분류하여 `feature`, `fix` 중 골라서 쓰는게 어떨까 하는 생각입니다.
모호한 표현이라는 느낌이 듭니다.

의견이 충분히 모이면 이번주 중으로 PR 작성하겠습니다.

**Comment by crowjdh** [comment]
1. 깃/깃허브의 기본 브렌치명이 master에서 main으로 제안된 시작은 [여기](https://lore.kernel.org/git/CAOAHyQwyXC1Z3v7BZAC+Bq6JBaM7FvBenA-1fcqeDV==apdWDg@mail.gmail.com/)서 확인할 수 있습니다.
  최근 IT업계에서 [PC주의](https://rms-support-letter.github.io/), [정치적 견해(반전 배너)](https://docs.nestjs.com/), 그와 관련된 [supply chain attack](https://web.archive.org/web/20220319151307/https://github.com/RIAEvangelist/peacenotwar/issues?q=is%3Aissue+is%3Aclosed) 등 **기술/프로덕트와 무관하게** 이념을 강요하는 풍조가 만연해 있어요.
  전 프로라면 무언가를 만드는 데에 이념이나 사상이 관여하지 않아야 한다고 생각하고, 이 변경은 근래의 큰 흐름의 결과물이라 생각하기 때문에 기존 마스터 브렌치를 유지하자는 의견입니다.
  그리고 깃허브에서는 main을 기본 브렌치명으로 정했지만 git은 아직 적용되지 않은 걸로 알고 있어요.

2. 좋은 것 같아요. 그럼 `enhancement`브랜치는 컨벤션에서 제거하게 되는 건가요?

**Reply by crowjdh** [reply]
아 적다 보니까 너무 드라이해 보이는데 제안은 감사합니다 ㅠ 다만 제가 알기론 기본 브렌치명은 저희가 픽스할 수 있는 걸로 알고 있어요. 저희가 일하는 입장에서는 실용성 없는 명명에 대해 고민하기 보다 변경이 미칠 영향(스크립트 호환성 등) 등에 더 집중하는게 맞다고 생각했어요 :)

**Comment by DonghyunNa** [comment]
1. 저도 이념이나 사상이 관여하지 않아야 한다는 생각에는 동의합니다. 같은 이유로 `main` 을 쓰면 어떨까 싶기는 합니다. main vs master 의 의미보다는, github의 기본 브랜치가 `main`이니, 저희가 이걸 비켜가면 오히려 말씀하신 스크립트 호환성 같은 문제를 좀 더 겪지 않을까 싶어, 이념이나 사상과는 관계없이 실용적으로 생각하여 하는 제안이긴 합니다.

2. `enhancement` 는 그 의미가 조금 모호하여, 제거 후 좀 더 그 의미에 맞게 `fix` 혹은 `feature`로 치환하여 사용하면 될 것 같다는 생각입니다.

**Reply by crowjdh** [reply]
1.
1.1 음 `github`의 기본 브랜치명은 `main`으로 바꼈지만 `git`의 기본 브렌치명은 아직 정해지지 않은 걸로 알고 있어요.([참조](https://git-scm.com/docs/git-init)) `main`이 `git`의 기본 브렌치명이 아닌 과도기에 굳이 넘어갈 필요가 없어 보여요.
1.2 github의 기본 브렌치명이 main으로 바껴 호환성이 걱정이 되는 거라면 [여기](https://github.com/organizations/QuesDevTeam/settings/repository-defaults)에서 기본 브렌치명을 바꾸는 걸로 쉽게 해결이 돼서 호환성문제 신경쓰지 않고 넘어갈 수 있을 것 같아요.

**Reply by DonghyunNa** [reply]
아 네 좋습니다. 업계동향을 좀 더 살핀 뒤, 나~중에 추후에 꼭 변경해야할 필요성이 느껴진다면, 다시 여기 제안 드리겠습니다!
